src/cargo/ops/resolve.rs

```diff
@@ -334,6 +340,24 @@ pub fn resolve_with_previous<'cfg>(
         warn,
     )?;
     resolved.register_used_patches(registry.patches());
+    if warn {
+        // It would be good if this warning was more targeted and helpful
+        // (such as showing close candidates that failed to match). However,
+        // that's not terribly easy to do, so just show a general help
+        // message.
+        let warnings: Vec<String> = resolved
+            .unused_patches()
+            .iter()
+            .map(|pkgid| format!("Patch for `{}` did not match any packages.", pkgid))
```

Comment:
I might recommend s/did not match any packages/did not end up being used in the crate graph/

(or something like that, it must match in one way or another but the error here is that it didn't end up actually being use)